src/layman/layer/filesystem/input_file.py

```diff
@@ -60,19 +48,20 @@ def get_layer_input_files(workspace, layername):
 
 
 def get_layer_info(workspace, layername):
-    abs_main_filepath = get_layer_main_file_path(workspace, layername, )
+    input_files = get_layer_input_files(workspace, layername, )
 
-    if abs_main_filepath is not None:
-        rel_main_filepath = os.path.relpath(abs_main_filepath, common_util.get_workspace_dir(workspace))
+    if input_files.saved_paths:
+        main_file_path = input_files.raw_or_archived_main_file_path or input_files.saved_paths[0]
```

Comment:
I do not see the case, where second part of `or` is needed. If it is, it should be part of some method in InputFiles.

Reply:
Solution is to add comment with description of such specific situation.